### PR TITLE
Update the instructions on setting up bors

### PR DESCRIPTION
**Pull request by rylev**
r? @pietroalbini

### CODE REVIEW

File: src/infra/docs/bors.md [hunk 1]
```diff
@@ -51,23 +48,43 @@ There are multiple steps needed to add a repository to our Bors instance:
...
-3. Add the repository name to the `permissions!` macro in the [team
+   These jobs need to run on specific branches (`auto` and `try`) so it's necessary
+   to add those branches to the list of branches tested by the CI provider. For GitHub
+   Actions that looks like this:
+
+   ```yaml
+   on:
+      push:
+          branches: [ 
+            master, # This is any branch you normally run CI on
```
Comment (ehuss) [inline src/infra/docs/bors.md:59]: Just curious, is there a particular reason to include `master` here?  Usually it shouldn't be necessary to run CI on master, since bors just ran the full test suite on the `auto` branch.  Unless you have special requirements (like `rust-lang/rust` which just does toolstate stuff on master), that essentially ends up running the same tests on the same commit twice.

File: src/infra/docs/bors.md [hunk 2]
```diff
@@ -21,10 +21,7 @@ failed are re-approved again on their own.
 
 There are multiple steps needed to add a repository to our Bors instance:
 
-1. The [@bors] GitHub account needs to be granted write access to the
-   repository.
```
Comment (pietroalbini) [inline src/infra/docs/bors.md:25]: Why was this removed? Bors still needs write access.

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: src/infra/docs/bors.md (hunk 1 above)
Comment (pietroalbini) [reply src/infra/docs/bors.md:59]: I think this is just an example, but `master` might be an unfortunate name (as it's one of the common default branches names, so people might expect the default branch has to be tested). Maybe replacing this with a commented `additional-branch`?

File: src/infra/docs/bors.md (hunk 2 above)
Comment (rylev) [reply src/infra/docs/bors.md:25]: I think I didn't remember needing to do this, but I'm probably misremembering

**Review by pietroalbini — APPROVED** [review]
